test/performance/scale_from_zero_test.go

```diff
@@ -0,0 +1,192 @@
+	if err := pkgTest.WaitForDeploymentState(
+		clients.KubeClient,
+		deploymentName,
+		isDeploymentScaledToZero(),
+		"DeploymentScaledToZero",
+		test.ServingNamespace,
+		2*time.Minute); err != nil {
+		return 0, fmt.Errorf("Failed waiting for deployment to scale to zero: %v", err)
+	}
+
+	start := time.Now()
+	logger.Info("Waiting for endpoint to serve request")
+	if _, err := pkgTest.WaitForEndpointState(
+		clients.KubeClient,
+		logger,
+		domain,
+		pkgTest.Retrying(pkgTest.MatchesBody(helloWorldExpectedOutput), http.StatusNotFound),
+		"HelloWorldServesText",
+		test.ServingFlags.ResolvableDomain); err != nil {
+		return 0, fmt.Errorf("The endpoint for Route %s at domain %s didn't serve the expected text \"%s\": %v", ro.Route.Name, domain, helloWorldExpectedOutput, err)
+	}
+
+	logger.Info("Request completed")
+	return time.Now().Sub(start), nil
+}
+
+func parallelScaleFromZero(t *testing.T, logger *logging.BaseLogger, count int) ([]time.Duration, error) {
+	ctx := context.TODO()
+	pc, err := Setup(ctx, logger, false)
+	if err != nil {
+		t.Fatalf("Failed to setup clients: %v", err)
+	}
+
+	testNames := make([]*test.ResourceNames, count)
+	durations := make([]time.Duration, count)
+
+	// Initialize our service names
+	for i := 0; i < count; i++ {
+		testNames[i] = &test.ResourceNames{
+			Service: test.AppendRandomString(serviceName, logger),
+			Image:   test.ImagePath("helloworld"),
+		}
+	}
+
+	cleanupNames := func() {
+		for i := 0; i < count; i++ {
+			if testNames[i] != nil {
+				TearDown(pc, logger, *testNames[i])
+			}
+		}
+	}
+	defer cleanupNames()
+	test.CleanupOnInterrupt(cleanupNames, logger)
+
+	g, _ := errgroup.WithContext(ctx)
+	for i := 0; i < count; i++ {
+		ndx := i
+		g.Go(func() error {
+			ro, err := test.CreateRunLatestServiceReady(logger, pc.E2EClients, testNames[ndx], &test.Options{})
+			if err != nil {
+				return fmt.Errorf("Failed to create Ready service: %v", err)
+			}
+			dur, err := runScaleFromZero(pc.E2EClients, logger, ro)
+			if err == nil {
+				durations[ndx] = dur
+			}
+			return err
+		})
+	}
+	err = g.Wait()
+
+	return durations, err
+}
+
+func getStats(durations []time.Duration) *stats {
+	var (
+		avg time.Duration
+	)
+
+	for _, dur := range durations {
+		avg += dur
+	}
+	avg = time.Duration(int64(avg) / int64(len(durations)))
+
+	return &stats{
+		avg: avg,
+	}
+}
+
+func testGrid(s *stats, tName string) error {
+	tc := []testgrid.TestCase{
+		{
+			ClassName: ScaleFromZeroTestGridClass,
+			Name:      fmt.Sprintf("%s/%s", ScaleFromZeroTestGridClass, tName),
+			Properties: testgrid.TestProperties{
+				Property: []testgrid.TestProperty{
+					{
+						Name:  ScaleFromZeroAvgTestGridProperty,
+						Value: float32(s.avg / (time.Second * 1000)),
+					},
+				},
+			},
+		},
+	}
+	return testgrid.CreateTestgridXML(tc, "TestPerformanceScaleFromZero")
+}
+
+func testScaleFromZero(t *testing.T, count int) {
+	logger := logging.GetContextLogger("TestScaleFromZero")
```

Comment:
Please create one logger per test by either creating it in the Test... methods or add the count to the logger name here. Will make it much easier to debug and find logs corresponding to tests later.